File: Programming Basics/C++/Nested loops/NumberPyramid/NumberPyramid.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cin >> n;

    int row = 1;
    int current = 1;

    while (current <= n)
    {
        for (int i = 0; i < row; i++)
        {
            cout << current << " ";
            if (++current > n) break;
        }

        row++;
        cout << endl;
    }
}

File: Programming Basics/JavaScript/Nested loops/Exercises/number-pyramid.js
function solve(input) {
    const n = Number(input[0]);

    let row = 1;
    let current = 1;

    while (current <= n) {
        let result = "";
        for (let i = 0; i < row; i++) {
            if (i > 0) result += " ";

            result += `${current}`;
            if (++current > n) break;
        }

        row++;
        console.log(result);
    }
}
